README.md
[![built with gulp](https://img.shields.io/badge/gulp-built_project-eb4a4b.svg?logo=data%3Aimage%2Fpng%3Bbase64%2CiVBORw0KGgoAAAANSUhEUgAAAAYAAAAOCAMAAAA7QZ0XAAAABlBMVEUAAAD%2F%2F%2F%2Bl2Z%2FdAAAAAXRSTlMAQObYZgAAABdJREFUeAFjAAFGRjSSEQzwUgwQkjAFAAtaAD0Ls2nMAAAAAElFTkSuQmCC)](http://gulpjs.com/)
[![GitHub license](https://img.shields.io/badge/license-GPLv2-blue.svg)](https://github.com/peiche/cover/blob/master/LICENSE.txt)
[![Build Status](https://travis-ci.org/peiche/cover2.svg?branch=master)](https://travis-ci.org/peiche/cover2)

# Cover2

Cover2 is a content-driven blogging theme for WordPress. Built on top of Automattic's Underscores, Cover2 allows you to focus on your writing. There are no sidebars to mess with, just a single column view of your content.

## Features

### Full-width featured images

When you use a featured image, it displays as a full-screen background image behind the title.

### Put widgets in their place

Cover2 puts your content first, exactly where it should be. But that doesn't mean you can't have widgets and menus, and that's where the overlay comes in.

The overlay is a full-screen menu and widget display. You can define one menu and as many widgets as your little heart desires.

Additionally, there are three footer widget positions, which collapse to a single column on small screens.

### Accent Color and Overlay Color Scheme

Customize Cover2's header color and overlay color scheme in the Customizer. Live preview shows you exactly what it'll look like before hitting Publish!

## Recommended Plugins

### Jetpack

Automattic's Jetpack plugin comes packed with modules for any theme to use, but Cover2 is designed to work nicely with these:

- **Site Logo**  
In the Customizer, you can not only set the site title and tagline, but also a site logo. You can enable and disable any combination of these three options.

- **Featured Posts**  
In the Customizer, you'll find the option to assign a specific tag to featured posts (the default is "featured"). Tagging a post will give it a special place on your blog's home page: it's displayed larger than the normal post listing, with its featured image displayed prominently behind it. If you have tagged more than one featured post, Cover2 will display them in a slider.

- **Infinite Scroll**  
The Infinite Scroll module already works just fine, and we're not messing with that. :)

### Aesop Story Engine

Cover2 was built from the ground up with Aesop Story Engine in mind. Break out of the content area with full-width components like images, galleries, maps, and more.

## Other Features

### Responsive Grid

Cover2 includes support for a mobile first 12-column grid. [Read about it here.](RESPONSIVE-GRID.md)

## Installation

### GitHub

You can download the latest from GitHub. Follow these steps to activate Cover2:

1. Download the [latest release](https://github.com/peiche/cover2/releases/latest).
2. In your admin panel, go to **Appearance > Themes** and click the **Add New** button.
3. Click **Upload** and **Choose File**, then select the theme's zip file. Click **Install Now**.
4. Click **Activate**.

## FAQ

1. **How do I set the background image behind the post title?**  
   When you use a featured image in Cover2, it displays as a full-screen background image behind the title.

2. **Can I change the font size?**  
   Cover2 does not allow you to change the default font size. I recommend creating a child theme before making changes to the theme.

3. **I am receiving an error in the Customizer, what should I do?**  
   Disable any caching plugins that you may have activated.

4. **How do I make links look like buttons?**

   A normal like looks like this:
   ```
   <a href="https://wordpress.org">WordPress</a>
   ```

   To make the link look like a gray button, add the `button` class, like this:
   ```
   <a href="https://wordpress.org" class="button">WordPress</a>
   ```

   To make the link look like a button with the same link color as defined in the Customizer, add the optional `default` class, like this:
   ```
   <a href="https://wordpress.org" class="button default">WordPress</a>
   ```

## Contribute

If you see something wrong, or you want to improve on what I've got here, feel free to submit an issue or create a pull request.

## License

Cover2 is [GPL v2.0 or later](LICENSE.txt).

Image used in screenshot.png - https://unsplash.com/photos/K3uOmmlQmOo by Angelina Litvin    
License - http://creativecommons.org/publicdomain/zero/1.0/  

All other resources are licensed as follows:

* [Headroom](http://wicky.nillia.ms/headroom.js/) - [MIT](https://github.com/WickyNilliams/headroom.js/blob/master/LICENSE)
* [FlexSlider](http://flexslider.woothemes.com) - [GPL v2.0 or later](https://github.com/woocommerce/FlexSlider/blob/master/LICENSE.md)
* [Bourbon Neat](http://neat.bourbon.io/) - [MIT](https://github.com/thoughtbot/neat/blob/master/LICENSE.md)
* [Nucleo Icons](https://nucleoapp.com/) - [Standard License](https://github.com/NucleoApp/license-standard)

## Inspiration

[Icon Fonts vs. SVG](https://css-tricks.com/icon-fonts-vs-svg/)  
[The 100% Correct Way To Do CSS Breakpoints](https://medium.freecodecamp.com/the-100-correct-way-to-do-css-breakpoints-88d6a5ba1862)  
[Reponsive Display Text](https://24ways.org/2016/responsive-display-text/)  
[Align SVG Icons to Text and Say Goodbye to Font Icons](https://blog.prototypr.io/align-svg-icons-to-text-and-say-goodbye-to-font-icons-d44b3d7b26b4#.yfiiz5rca)  
[Your Body Text Is Too Small](https://blog.marvelapp.com/body-text-small/)  


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. This project adheres to [Semantic Versioning](http://semver.org/).

**1.4.2 - 12/10/2018**
- Fixed fatal error in WordPress 5.0. ([#37](https://github.com/peiche/cover2/issues/37), [#38](https://github.com/peiche/cover2/issues/38))
- Fixed inconsistent font weights.
- Fixed styles for Gutenberg image blocks and captions.
- Updated Gutenberg pullquote styles.
- Updated post display specific to quote post format.
- Updated tag cloud widget style.
- Added styles for [Editor Blocks](https://editorblockswp.com/) plugin.
- Added styles for [Atomic Blocks](https://atomicblocks.com/) plugin.
- Added support for larger featured image size in Algolia search results.
- Added utility class styles.
- Added Jetpack social menu to footer.
- Added Customizer setting to display featured images on post listings.
- Added form styles.
- Added calendar icon to post metadata.
- Added SVG support to Jetpack social menu.
- Added class support to icon utility function.
- Added missing icons to Jetpack social menu:
  - [Bitbucket](https://bitbucket.org/)
  - [DEV](https://dev.to/)
  - [GitLab](https://about.gitlab.com/)
  - [Sourcerer](https://sourcerer.io/)
  - [StackShare](https://stackshare.io/) (Note: the StackShare icon is currently broken when using a dark footer.)
- Disabled autocomplete on search form.

**1.4.1 - 8/28/2018**
- Fixed non-printable character on Algolia search result template.
- Fixed site logo aspect ratio.
- Fixed line width for navbar icons.
- Removed static posts Customizer setting.
- Removed most floats and clears in favor of Flexbox.
- Changed panel tag from article to section.
- Updated bookmark icon in navbar with [Nucleo's icon transition tool](https://nucleoapp.com/tool/icon-transition).
- Updated Aesop Story Engine detection when Gutenberg plugin is active.
- Updated Algolia search result page's current page style color.
- Added recent posts logic if the blog page is assigned to a front page panel.
- Added dark and light icons for Jetpack image carousel.
- Added gallery post format support.

**1.4.0 - 8/3/2018**
- Fixed button styles in header, navigation, and overlay.
- Fixed top margin for paged blog posts.
- Fixed Jetpack embed styles when Gutenberg is active.
- Updated Aesop Story Engine plugin's lightbox icons to match Cover2 style.
- Updated Gallery post format icon.
- Updated Jetpack plugin's Contact Form module styles.
- Updated Jetpack plugin's Carousel module styles.
- Updated blockquote styles (Gutenberg and otherwise).
- Updated Gutenberg embed styles.
- Updated Gutenberg cover image text style.
- Adjusted header image opacity to increase legibility.
- Replaced Algolia PNG logo with SVG.
- Rewrote responsive grid to use Flexbox.
- Added styles for Aesop Story Engine plugin's collection component.
- Added styles for Jetpack plugin's contact form module.
- Added panel options to static front page.
- Added recent post options to static front page.
- Added featured post options to static front page (requires Jetpack plugin).
- Added multiple tags icon.
- Added custom header support for static home page.
- Added styles for Jetpack's Related Posts module's grid and list views.
- Added Gutenberg paragraph font size styles.
- Added Gutenberg custom colors.
- Added author name and link to Algolia search results template.

**1.3.1 - 4/20/2018**
- Fixed content and button line-height to address [issue 30](https://github.com/peiche/cover2/issues/30) (again).
- Fixed typo in Jetpack featured content description.
- Added support for button classes on menu items.

**1.3.0 - 4/9/2018**
- Fixed posts navigation margin.
- Fixed blog page top margin.
- Fixed comment navigation width.
- Fixed footer widget colors.
- Fixed word break and hyphenation styles.
- Fixed comment form field widths.
- Fixed full-height header to address [issue 31](https://github.com/peiche/cover2/issues/31).
- Changed header styles.
- Changed posts and comment navigation button styles.
- Changed Customizer footer color options.
- Added support for new [Jetpack social media icons widget](https://jetpack.com/2018/04/03/jetpack-6-0/).
- Added Customizer option to use accent color in theme icons.
- Added Customizer option to add configurable front page content.
- Replaced old grid styles with 12-column responsive grid. [Read the documentation.](RESPONSIVE-GRID.md)

**1.2.2 - 3/16/2018**
- Fixed broken close video button for Featured Video Plus.
- Fixed posts navigation vertical alignment on small screens to address [issue 28](https://github.com/peiche/cover2/issues/28).
- Updated Nucleo icons and added accent color.
- Added background color transition to Jetpack's social network icons.

**1.2.1 - 3/13/2018**
- Fixed archive header margin to address [issue 26](https://github.com/peiche/cover2/issues/26).
- Fixed paged post button styles.
- Fixed nav button focus and hover states.
- Added brand colors to Jetpack social media menu.
- Split Jetpack, Aesop Story Engine, and Algolia styles out of main stylesheet.

**1.2.0 - 2/8/2018**
- Fixed logo scaling (if SVG is used) in Internet Explorer.
- Updated button styles. Buttons now default to "solid" style, and support an optional "ghost" class for an outlined style.
- Tweaked header divider and excerpt widths.
- Added filter sidebar to Algolia Instant Search template. Small screens display a filter button in the header.

**1.1.3 - 2/4/2018**
- Fixed "posted on" and "byline" strings to address [issue 22](https://github.com/peiche/cover2/issues/22).

**1.1.2 - 1/30/2018**
- Fixed Jetpack infinite footer style.
- Fixed footer widget alignment.
- Fixed paging link styles.
- Fixed translation logic for categories and tags.
- Updated icons for video and gallery post formats.

**1.1.1 - 1/26/2018**
- Tweaked footer style.
- Updated styles for Aesop Story Engine map component.
- Updated styles for Gutenberg cover image and button blocks.

**1.1.0 - 1/10/2018**
- Fixed broken pages widget.

functions.php
<?php
/**
 * Components functions and definitions.
 *
 * @link https://developer.wordpress.org/themes/basics/theme-functions/
 *
 * @package Cover2
 */

if ( ! function_exists( 'cover2_setup' ) ) :
/**
 * Sets up theme defaults and registers support for various WordPress features.
 *
 * This function is hooked into the aftercomponentsetup_theme hook, which
 * runs before the init hook. The init hook is too late for some features, such
 * as indicating support for post thumbnails.
 */
function cover2_setup() {
	/*
	 * Make theme available for translation.
	 * Translations can be filed in the /languages/ directory.
	 * If you're building a theme based on components, use a find and replace
	 * to change 'cover2' to the name of your theme in all the template files.
	 */
	load_theme_textdomain( 'cover2', get_template_directory() . '/languages' );

	// Add default posts and comments RSS feed links to head.
	add_theme_support( 'automatic-feed-links' );

	/*
	 * Let WordPress manage the document title.
	 * By adding theme support, we declare that this theme does not use a
	 * hard-coded <title> tag in the document head, and expect WordPress to
	 * provide it for us.
	 */
	add_theme_support( 'title-tag' );

	/*
	 * Enable support for Post Thumbnails on posts and pages.
	 *
	 * @link https://developer.wordpress.org/themes/functionality/featured-images-post-thumbnails/
	 */
	add_theme_support( 'post-thumbnails' );

	add_image_size( 'cover2-featured-image', 1600, 9999 );
	add_image_size( 'cover2-large', 2000, 1500, true );

	// This theme uses wp_nav_menu() in one location.
	register_nav_menus( array(
		'top'		=> esc_html__( 'Top Menu', 'cover2' ),
		'mini'		=> esc_html__( 'Mini Menu', 'cover2' ),
	) );

	/**
	 * Add support for core custom logo.
	 */
	add_theme_support( 'custom-logo', array(
		'height'      => 200,
		'width'       => 200,
		'flex-width'  => true,
		'flex-height' => true,
	) );
	
	/**
	 *  Add support for custom video header.
	 */
	add_theme_support( 'custom-header' );
	
	/*
	 * Switch default core markup for search form, comment form, and comments
	 * to output valid HTML5.
	 */
	add_theme_support( 'html5', array(
		'search-form',
		'comment-form',
		'comment-list',
		'gallery',
		'caption',
	) );

	/*
	 * Enable support for Post Formats.
	 * See https://developer.wordpress.org/themes/functionality/post-formats/
	 */
	add_theme_support( 'post-formats', array(
		'image',
		'video',
		'audio',
		'quote',
		'aside',
		'link',
		'chat',
		'status',
		'gallery',
	) );

	/*
	 * Enable support for widget selective refresh.
	 * See https://make.wordpress.org/core/2016/03/22/implementing-selective-refresh-support-for-widgets/
	 */
	add_theme_support( 'customize-selective-refresh-widgets' );

	/*
	 * Enable support for the Gutenberg editor.
	 * See http://gutenberg-devdoc.surge.sh/reference/theme-support/
	 */
	add_theme_support( 'align-wide' );
	
	// Matches color map defined in /assets/stylesheets/variables/_colors.scss
	// Uses accent color defined in Customizer
	add_theme_support( 'editor-color-palette', array(
	    array(
	        'name' => __( 'accent', 'cover2' ),
	        'slug' => 'accent',
	        'color' => 'hsl(' . absint( get_theme_mod( 'accent_color', 250 ) ) . ', 75%, 50%)',
	    ),
	    array(
	        'name' => __( 'dark', 'cover2' ),
	        'slug' => 'dark',
	        'color' => '#404040',
	    ),
	    array(
	        'name' => __( 'light', 'cover2' ),
	        'slug' => 'light',
	        'color' => '#eee',
	    ),
	    array(
	        'name' => __( 'white', 'cover2' ),
	        'slug' => 'white',
	        'color' => '#fff',
	    ),
	) );
	add_theme_support( 'editor-font-sizes', array(
	    array(
	        'name' => __( 'small', 'cover2' ),
	        'shortName' => __( 'S', 'cover2' ),
	        'size' => 16,
	        'slug' => 'small'
	    ),
	    array(
	        'name' => __( 'regular', 'cover2' ),
	        'shortName' => __( 'M', 'cover2' ),
	        'size' => 20,
	        'slug' => 'regular'
	    ),
	    array(
	        'name' => __( 'large', 'cover2' ),
	        'shortName' => __( 'L', 'cover2' ),
	        'size' => 24,
	        'slug' => 'large'
	    ),
	    array(
	        'name' => __( 'larger', 'cover2' ),
	        'shortName' => __( 'XL', 'cover2' ),
	        'size' => 30,
	        'slug' => 'larger'
	    )
	) );

	/*
	 * Enable support for excerpts on pages.
	 */
	add_post_type_support( 'page', 'excerpt' );
}
endif;
add_action( 'after_setup_theme', 'cover2_setup' );

/**
 * Set the content width in pixels, based on the theme's design and stylesheet.
 * We're giving it an extra large size to accommodate large screens.
 *
 * Priority 0 to make it available to lower priority callbacks.
 *
 * @global int $content_width
 */
function cover2_content_width() {
	$GLOBALS['content_width'] = apply_filters( 'cover2_content_width', 1600 );
}
add_action( 'after_setup_theme', 'cover2_content_width', 0 );

/**
 * Register custom fonts.
 */
function cover2_fonts_url() {
	$fonts_url = '';

	/**
	 * Translators: If there are characters in your language that are not
	 * supported by these fonts, translate this to 'off'. Do not translate
	 * into your own language.
	 */
	$monserrat = _x( 'on', 'Montserrat font: on or off', 'cover2' );
	$open_sans = _x( 'on', 'Open Sans font: on or off', 'cover2' );
	$source_code_pro = _x( 'on', 'Source Code Pro font: on or off', 'cover2' );

	if ( 'off' !== $monserrat && 'off' !== $open_sans && 'off' !== $source_code_pro ) {
		$font_families = array();

		$font_families[] = 'Montserrat:600|Open+Sans:300,300i,600|Source+Code+Pro:300';

		$query_args = array(
			'family' => urlencode( implode( '|', $font_families ) ),
			'subset' => urlencode( 'latin' ),
		);

		$fonts_url = add_query_arg( $query_args, 'https://fonts.googleapis.com/css' );
	}

	return esc_url_raw( $fonts_url );
}

/**
 * Add preconnect for Google Fonts.
 *
 * @since Cover2 1.0
 *
 * @param array  $urls           URLs to print for resource hints.
 * @param string $relation_type  The relation type the URLs are printed.
 * @return array $urls           URLs to print for resource hints.
 */
function cover2_resource_hints( $urls, $relation_type ) {
	if ( wp_style_is( 'cover2-fonts', 'queue' ) && 'preconnect' === $relation_type ) {
		$urls[] = array(
			'href' => 'https://fonts.gstatic.com',
			'crossorigin',
		);
	}

	return $urls;
}
add_filter( 'wp_resource_hints', 'cover2_resource_hints', 10, 2 );

/**
 * Register widget area.
 *
 * @link https://developer.wordpress.org/themes/functionality/sidebars/#registering-a-sidebar
 */
function cover2_widgets_init() {
	// Overlay widget.
	register_sidebar( array(
		'name'          => esc_html__( 'Overlay', 'cover2' ),
		'id'            => 'sidebar-overlay',
		'description'   => __( 'Add widgets here to appear in your sidebar.', 'cover2' ),
		'before_widget' => '<section id="%1$s" class="widget %2$s">',
		'after_widget'  => '</section>',
		'before_title'  => '<h2 class="widget-title">',
		'after_title'   => '</h2>',
	) );
	
	// Overlay widget.
	register_sidebar( array(
		'name'          => esc_html__( 'Footer (Left)', 'cover2' ),
		'id'            => 'sidebar-footer-1',
		'description'   => __( 'Add widgets here to appear in your sidebar.', 'cover2' ),
		'before_widget' => '<section id="%1$s" class="widget %2$s">',
		'after_widget'  => '</section>',
		'before_title'  => '<h2 class="widget-title">',
		'after_title'   => '</h2>',
	) );
	register_sidebar( array(
		'name'          => esc_html__( 'Footer (Middle)', 'cover2' ),
		'id'            => 'sidebar-footer-2',
		'description'   => __( 'Add widgets here to appear in your sidebar.', 'cover2' ),
		'before_widget' => '<section id="%1$s" class="widget %2$s">',
		'after_widget'  => '</section>',
		'before_title'  => '<h2 class="widget-title">',
		'after_title'   => '</h2>',
	) );
	register_sidebar( array(
		'name'          => esc_html__( 'Footer (Right)', 'cover2' ),
		'id'            => 'sidebar-footer-3',
		'description'   => __( 'Add widgets here to appear in your sidebar.', 'cover2' ),
		'before_widget' => '<section id="%1$s" class="widget %2$s">',
		'after_widget'  => '</section>',
		'before_title'  => '<h2 class="widget-title">',
		'after_title'   => '</h2>',
	) );
}
add_action( 'widgets_init', 'cover2_widgets_init' );

/**
 * Display custom color CSS.
 */
function cover2_colors_css_wrap() {
	require_once( get_parent_theme_file_path( '/inc/color-patterns.php' ) );
	$hue = absint( get_theme_mod( 'accent_color', 250 ) );
?>
	<style type="text/css" id="custom-theme-colors" <?php if ( is_customize_preview() ) { echo 'data-hue="' . $hue . '"'; } ?>>
		<?php echo cover2_custom_colors_css(); ?>
	</style>
	<meta name="theme-color" content="hsl(<?php echo $hue; ?>, 75%, 50%)">
<?php }
add_action( 'wp_head', 'cover2_colors_css_wrap' );

/**
 * Enqueue scripts and styles.
 */
function cover2_scripts() {

	// Add custom fonts, used in the main stylesheet.
	wp_enqueue_style( 'cover2-fonts', cover2_fonts_url(), array(), null );

	// Theme stylesheet.
	wp_register_style( 'cover2-style', get_stylesheet_uri(), array(), filemtime( get_template_directory() . '/style.css' ) );
	wp_enqueue_style( 'cover2-style' );

	wp_enqueue_script( 'cover2-headroom', get_template_directory_uri() . '/dist/js/headroom.min.js', array( 'jquery' ), filemtime( get_template_directory() . '/dist/js/headroom.min.js' ), true );

	wp_enqueue_script( 'cover2-main', get_template_directory_uri() . '/dist/js/main.js', array( 'jquery' ), filemtime( get_template_directory() . '/dist/js/main.js' ), true );
	wp_localize_script( 'cover2-main', 'menuToggleText', array(
		'open'   => esc_html__( 'Open child menu', 'cover2' ),
		'close'  => esc_html__( 'Close child menu', 'cover2' ),
		'icon'   => cover2_get_svg( array( 'icon' => 'caret-down', 'fallback' => true ) ),
	) );

	wp_enqueue_script( 'cover2-navigation', get_template_directory_uri() . '/dist/js/navigation.js', array(), filemtime( get_template_directory() . '/dist/js/navigation.js' ), true );

	wp_enqueue_script( 'cover2-skip-link-focus-fix', get_template_directory_uri() . '/dist/js/skip-link-focus-fix.js', array(), filemtime( get_template_directory() . '/dist/js/skip-link-focus-fix.js' ), true );

	if ( is_singular() && comments_open() && get_option( 'thread_comments' ) ) {
		wp_enqueue_script( 'comment-reply' );
	}

	if ( cover2_has_multiple_featured_posts() ) {
		wp_enqueue_script( 'flexslider', get_template_directory_uri() . '/dist/js/jquery.flexslider-min.js', array( 'jquery' ), filemtime( get_template_directory() . '/dist/js/jquery.flexslider-min.js' ), true );
		wp_enqueue_script( 'cover2-flexslider', get_template_directory_uri() . '/dist/js/flexslider.js', array( 'flexslider' ), filemtime( get_template_directory() . '/dist/js/flexslider.js' ), true );
	}

	// Featured Video Plus plugin compatibility.
	if ( function_exists( 'has_post_video' ) && has_post_video() && (is_single() || is_page()) ) {
		wp_enqueue_script( 'vimeo', get_template_directory_uri() . '/dist/js/player.min.js', array(), filemtime( get_template_directory() . '/dist/js/player.min.js' ), true );
		wp_enqueue_script( 'cover2-featured-video-plus', get_template_directory_uri() . '/dist/js/featured-video-plus.js', array(), filemtime( get_template_directory() . '/dist/js/featured-video-plus.js' ), true );
	}
}
add_action( 'wp_enqueue_scripts', 'cover2_scripts' );

/**
 * Custom template tags for this theme.
 */
require get_template_directory() . '/inc/template-tags.php';

/**
 * Custom functions that act independently of the theme templates.
 */
require get_template_directory() . '/inc/extras.php';

/**
 * Customizer additions.
 */
require get_template_directory() . '/inc/customizer.php';

/**
 * Jetpack compatibility file.
 */
require get_template_directory() . '/inc/jetpack.php';

/**
 * SVG icons functions and filters.
 */
require get_parent_theme_file_path( '/inc/icon-functions.php' );

/**
 * Load Aesop Story Engine compatibility.
 */
require get_template_directory() . '/inc/aesop.php';

/**
 * Load Algolia compatibility.
 */
require get_template_directory() . '/inc/algolia.php';

/**
 * Load Atomic Blocks compatibility.
 */
require get_template_directory() . '/inc/atomic-blocks.php';

/**
 * Load Editor Blocks compatibility.
 */
require get_template_directory() . '/inc/editor-blocks.php';
